### PR TITLE
feat: add ability to customize API root path in service URL

### LINKED ISSUE #105: Model selection field does not show up when trying to use perplexity openai endpoint
### Which version of integration_openai are you using?

2.0.1

### Which version of Nextcloud are you using?

v29.0.2

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

126.0.1 (64-bit)

### Describe the Bug

I try to configure the openai endpoint of perplexity, bit the model selection just does not show up.
nextcloud_perplexity

### Expected Behavior

I can select the model.

### To Reproduce

Try to use perplexity as ai endpoint.

### PR DESCRIPTION
**Pull request by edward-ly**
Resolves #105.

### CODE REVIEW

File: src/components/AdminSettings.vue [hunk 1]
```diff
@@ -19,15 +19,15 @@
 						<EarthIcon />
 					</NcTextField>
 					<NcButton type="tertiary"
-						:title="t('integration_openai', 'Leave empty to use {openaiApiUrl}', { openaiApiUrl: 'https://api.openai.com' })">
+						:title="t('integration_openai', 'Leave empty to use {openaiApiUrl}', { openaiApiUrl: 'https://api.openai.com/v1' })">
```
Comment (kyteinsky) [inline src/components/AdminSettings.vue:22]: let's add v1 to line 15 too

File: src/components/AdminSettings.vue [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
 						</template>
 					</NcButton>
 				</div>
 				<NcNoteCard type="info">
 					<p>
-						{{ t('integration_openai', 'This should be the address of your LocalAI instance (or any service implementing an API similar to OpenAI). This URL will be accessed by your Nextcloud server.') }}
+						{{ t('integration_openai', 'This should include the address of your LocalAI instance (or any service implementing an API similar to OpenAI) along with the root path of the API. This URL will be accessed by your Nextcloud server.') }}
 					</p>
 					<p>
 						{{ t('integration_openai', 'This can be a local address with a port like {example}. In this case make sure \'allow_local_remote_servers\' is set to true in config.php', { example : 'http://localhost:8080' }) }}
```
Comment (kyteinsky) [inline src/components/AdminSettings.vue:33]: same here, v1 should be appended to the url

**Comment by kyteinsky** [comment]
Adding a preview like a computed URL for an example endpoint like chat completions `http://localhost:8080/v1/chat/completions` when the service url field is modified would be convenient as Julien said here: https://github.com/nextcloud/integration_openai/issues/105#issuecomment-2396639019

**Review by julien-nc — CHANGES_REQUESTED** [review]
I agree with the review from @kyteinsky . Appart from that, all good.
For appconfig values, using IAppConfig in the migration step was the right thing to do :+1:

**Comment by edward-ly** [comment]
Done. I decided to use another NcNoteCard to display the preview as I thought NcTextField's helperText in dark mode was too faint to be usable, but I am open to other ideas.

**Review by kyteinsky — APPROVED** [review]

**Review by julien-nc — APPROVED** [review]
:+1:

**Pull request merged**